windows/drivers/uwb/simulator/UwbSimulatorDdiHandler.cxx

```diff
@@ -45,11 +45,11 @@ UwbSimulatorDispatchEntryLrp (*MakeLrpDispatchEntry)(ULONG, typename UwbSimulato
 const std::initializer_list<UwbSimulatorDispatchEntry<UwbSimulatorDdiHandler>> UwbSimulatorDdiHandler::Dispatch{
     // GUID_UWB_DEVICE_INTERFACE Handlers
     MakeLrpDispatchEntry<UWB_DEVICE_RESET, UWB_STATUS>(IOCTL_UWB_DEVICE_RESET, &UwbSimulatorDdiHandler::OnUwbDeviceReset),
-    MakeLrpDispatchEntry<Unrestricted, Unrestricted>(IOCTL_UWB_GET_DEVICE_INFO, &UwbSimulatorDdiHandler::OnUwbGetDeviceInformation),
-    MakeLrpDispatchEntry<Unrestricted, Unrestricted>(IOCTL_UWB_GET_DEVICE_CAPABILITIES, &UwbSimulatorDdiHandler::OnUwbGetDeviceCapabilities),
+    MakeLrpDispatchEntryWithSize(IOCTL_UWB_GET_DEVICE_INFO, &UwbSimulatorDdiHandler::OnUwbGetDeviceInformation, 0, offsetof(UWB_DEVICE_INFO, vendorSpecificInfo)),
+    MakeLrpDispatchEntryWithSize(IOCTL_UWB_GET_DEVICE_CAPABILITIES, &UwbSimulatorDdiHandler::OnUwbGetDeviceCapabilities, 0, offsetof(UWB_DEVICE_CAPABILITIES, capabilityParams)),
     MakeLrpDispatchEntry<UWB_SET_DEVICE_CONFIG_PARAMS, Unrestricted>(IOCTL_UWB_GET_DEVICE_CONFIG_PARAMS, &UwbSimulatorDdiHandler::OnUwbGetDeviceConfigurationParameters),
     MakeLrpDispatchEntry<UWB_GET_DEVICE_CONFIG_PARAMS, Unrestricted>(IOCTL_UWB_SET_DEVICE_CONFIG_PARAMS, &UwbSimulatorDdiHandler::OnUwbSetDeviceConfigurationParameters),
-    MakeLrpDispatchEntry<UWB_GET_APP_CONFIG_PARAMS, Unrestricted>(IOCTL_UWB_GET_APP_CONFIG_PARAMS, &UwbSimulatorDdiHandler::OnUwbGetApplicationConfigurationParameters),
+    MakeLrpDispatchEntryWithSize(IOCTL_UWB_GET_APP_CONFIG_PARAMS, &UwbSimulatorDdiHandler::OnUwbGetApplicationConfigurationParameters, offsetof(UWB_GET_APP_CONFIG_PARAMS, appConfigParams), offsetof(UWB_APP_CONFIG_PARAMS, appConfigParams)),

```

Comment:
These will require changes to the handler functions (listed in each dispatch entry here) to provide partial responses. Right now, they'll only populate the output buffer if the response is large enough. Shouldn't be too much work. The handler for `IOCTL_UWB_SET_APP_CONFIG_PARAMS` (`OnUwbSetApplicationConfigurationParameters`) does this correctly, so you can use that as a template.

Reply:
https://github.com/microsoft/nearobject-framework/blob/f10adf9f05e2b242f594087c29b9cf35c9a22657/windows/drivers/uwb/simulator/UwbSimulatorDdiHandler.cxx#L258-L275

Reply:
I updated the handlers to follow the `OnUwbSetApplicationConfigurationParameters` format but did not test them, so this may require a closer look

Reply:
Let's test them please so we know whether the updated code works.

Reply:
I can confirm that the changes in `OnUwbGetApplicationConfigurationParameters` work, which I would consider to be the "most complicated" example out of the 3 changes. 

The current functionality in `OnUwbGetDeviceInformation` still works, although we currently don't test the variable size array here (vendor specific info).

We don't have a raw command for device capabilities, so that one is untested.

Reply:
Thanks for doing the tests. We don't need to worry about the ones that aren't exercised (ie. those without raw commands from the cli), so no worries there.

If you can confirm `OnUwbGetApplicationConfigurationParameters` works with the simulator driver, then we're good to go.